### DISCUSSION TITLE
MPEG-DASH plug-in implementation

### DISCUSSION THREAD

**Discussion by phpb-com** [shiyiya/oplayer#15] [Ideas]
Let’s use discussions to talk about the plug-in implementation. We can also put other topics here, I guess. @shiyiya so you are aware of this thread 👍🏻

**Comment by shiyiya** [comment]
👌

**Comment by shiyiya** [comment]
Any progress? @phpb-com

**Comment by phpb-com** [comment]
Not today, no time to spare. Sorry.

**Comment by shiyiya** [comment]
Anytime, thanks!

**Comment by phpb-com** [comment]
I should have something workable by this Sunday, no time these weekdays

**Comment by phpb-com** [comment]
Quick update, since there is an issue with dashjs and SSR, where it requires window to be defined, it is not possible to use it like hls.js. I am abandoning the idea for now until I know of a fix or SSR friendly implementation appears. I did look into rx-player which does DASH but it is huge and probably not the best option.

**Comment by shiyiya** [comment]
If I have time I will try to implement it and will let you know here if there is progress

**Comment by shiyiya** [comment]
https://github.com/shiyiya/oplayer/tree/main/packages/dash

**Comment by phpb-com** [comment]
Wow, just gave it a quick test with SSR and everything, so far no errors. I'll keep you posted on this thread about testing and config. Excellent work!

**Comment by phpb-com** [comment]
I had one question, would it be possible to pass multiple URLs to the player in the order of preference, so it can try to load the first one, then continue falling until it can? Something like:
```tsx
onst playerProps: ReactOPlayerProps = {
    source: [{ src: 'https://example.com/video.mpd', poster: 'https://example.com/video-poster.jpg', format: 'dash' },
                   { src: 'https://example.com/video.m3u8', poster: 'https://example.com/video-poster.jpg', format: 'm3u8' },
                   { src: 'https://example.com/video.mp4', poster: 'https://example.com/video-poster.jpg', format: 'mp4' }]
    preload: 'auto', // Keep auto or iPhone stalls
    playsinline: true,
}
```

**Reply by shiyiya** [reply]
https://github.com/Enime-Project/enime.moe/blob/master/components/player/index.vue#L72

I'm not going to implement it, I don't think it's necessary.

**Comment by phpb-com** [comment]
No problems, I was just thinking of switching between dash and hls, depending on the client. I sure can do it in my own code. Thanks.
